### PR TITLE
Adding production if/else check to use ECR public from staging for packages helm chart.

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/pkg/generate_spec.go [hunk 1]
```diff
@@ -492,6 +492,23 @@ func (r *ReleaseConfig) GetSourceImageURI(name, repoName string, tagOptions map[
 
 func (r *ReleaseConfig) GetSourceHelmURI(repoName string) (string, error) {
 	var sourceImageUri string
+	// If it's running a production release we only lookup from ECR public of the staging account.
+	if r.ReleaseEnvironment == "production" {
+		ecrpublicClient, err := NewECRPublicClient()
```
Comment (abhay-krishna) [inline release/pkg/generate_spec.go:497]: We probably don't need to create new clients. The release config struct has source clients and release clients configured in it, for ecr, ecr public and s3

Comment (jonahjon) [reply release/pkg/generate_spec.go:497]: This is using the SDK v2 so we might need to do so.

File: release/pkg/assets_package_controller.go [hunk 2]
```diff
@@ -44,7 +44,14 @@ func (r *ReleaseConfig) GetPackagesAssets() ([]Artifact, error) {
 	if err != nil {
 		return nil, errors.Cause(err)
 	}
-	sourceHelmURI, err := r.GetSourceHelmURI(repoName)
+	var sourceHelmURI string
```
Comment (ewollesen) [inline release/pkg/assets_package_controller.go:47]: Is the value returned by `GetSourceHelmURI` incorrect for a production case? Is this the correct place to handle this branching? Or should the environment be passed to `GetShourceHelmURI`?

Comment (jonahjon) [reply release/pkg/assets_package_controller.go:47]: The production role doesn't have access to call `GetSourceHelmURI`, and it's not needed so we want to skip that all together in that scenario.

Comment (jonahjon) [reply release/pkg/assets_package_controller.go]: sounds good, fixed this to the way you mentioned.

File: release/pkg/generate_spec.go [hunk 3]
```diff
@@ -490,22 +490,29 @@ func (r *ReleaseConfig) GetSourceImageURI(name, repoName string, tagOptions map[
...
-func (r *ReleaseConfig) GetSourceHelmURI(repoName string) (string, error) {
-	var sourceImageUri string
-	ecrClient, err := NewECRClient()
-	if err != nil {
-		return "", err
-	}
-	latestTag, err := ecrClient.GetLatestUploadHelmSha(repoName)
-	if err != nil {
-		return "", err
+//GetSourceHelmURI is a function which finds the latest helm chart tag for a given repository, and returns the full URI including that tag.
```
Comment (ewollesen) [inline release/pkg/generate_spec.go:493]:
```suggestion
// GetSourceHelmURI returns the URI for the latest tagged helm chart in the given repository.
```

**Review by ewollesen — APPROVED** [review]
/lgtm

**Pull request merged**